crates/lox_core/src/time/continuous/deltas.rs

```diff
@@ -0,0 +1,168 @@
+/*
+ * Copyright (c) 2024. Helge Eichhorn and the LOX contributors
+ *
+ * This Source Code Form is subject to the terms of the Mozilla Public
+ * License, v. 2.0. If a copy of the MPL was not distributed with this
+ * file, you can obtain one at https://mozilla.org/MPL/2.0/.
+ */
+
+use num::traits::Inv;
+use num::ToPrimitive;
+
+use crate::time::constants::f64;
+use crate::time::constants::u128;
+
+/// An absolute continuous time difference with femtosecond precision.
+#[derive(Debug, Default, Copy, Clone, Eq, PartialEq)]
+pub struct TimeDelta {
+    pub seconds: u64,
+    pub femtoseconds: u64,
+}
+
+impl TimeDelta {
+    pub fn int_seconds(seconds: u64) -> Self {
+        Self {
+            seconds,
+            femtoseconds: 0,
+        }
+    }
+
+    pub fn seconds(value: f64) -> Self {
+        if value < f64::FEMTOSECONDS_PER_SECOND.inv() || value.is_nan() {
+            return Self::default();
+        }
+        if value > u64::MAX as f64 {
```

Comment:
My first version of this returned an `Option<TimeDelta>` but I found this unsatisfactory from an ergonomics perspective. This would introduce a lot of error checking for an edge case that is very hard to trigger accidentally in practice (create a `TimeDelta` larger than 500 billion years). In addition, it does not solve the next problem that depending on the size of the `Time` and `TimeDelta` the resulting `Time` could also saturate when the two are added together.
Thus, I would propose to accept that `Time` and `TimeDelta` are saturating and clearly document this.

Reply:
Great idea!